src/silx/gui/plot/ImageView.py

```diff
@@ -756,7 +756,7 @@ def setImage(self, image, origin=(0, 0), scale=(1., 1.),
 
         data = numpy.array(image, order='C', copy=copy)
         assert data.size != 0
-        assert len(data.shape) == 2
+        assert data.ndim in (2, 3)
```

Comment:
Maybe catch early RGB(A) without the right number of channels:
```suggestion
        assert data.ndim == 2 or (data.ndim == 3 and data.shape[2] in (3, 4))
```